addons/account/models/account_reconcile_model.py

```diff
@@ -25,7 +25,7 @@ class AccountReconcileModel(models.Model):
 
     # ===== Conditions =====
     match_journal_ids = fields.Many2many('account.journal', string='Journals',
-        domain="[('type', 'in', ('bank', 'cash'))]",
+        domain="['&',('type', 'in', ('bank', 'cash')), ('company_id', '=', company_id)]",
```

Comment:
Just some side-note: you're not obliged to give the '&' in this case. It'll be implicit if you say nothing :) (usually, we don't put it)